src/Session.cc

```diff
@@ -215,21 +219,27 @@ bool Session::FillExtendedFileInfo(CARTA::FileInfoExtended& extended_info, CARTA
 
             if (hdu.empty() && (file_info.type() == CARTA::FileType::FITS)) {
                 // File info adds empty string for FITS
-                std::vector<std::string> hdu_list;
-                std::string message;
-                FitsHduList fits_hdu_list(fullname);
-                fits_hdu_list.GetHduList(hdu_list, message);
+                if (IsCompressedFits(fullname)) {
+                    std::map<std::string, CARTA::FileInfoExtended> hdu_info_map;
+                    file_info_ok = ext_info_loader.FillFitsFileInfoMap(hdu_info_map, fullname, message);
```

Comment:
This works but is a lot of wasted effort to fill FileInfoExtended header_entries and computed_entries for all HDUs just to throw away everything but the first HDU string.  We could modify `CompressedFits::GetFitsHeaderInfo` (e.g. add arguments `int& first_hdu, bool first_hdu_only=false`, set to true for this), or create a similar much smaller function `int CompressedFits::GetFirstImageHdu()`, which stops after IsImageHdu returns true and returns the hdu.

I could do this if you'd rather not work with zlib, but it's really just adding gzclose and returning the hdu number when you find the first image HDU.

Reply:
Also check whether there is an image HDU detected, perhaps there are FITS files with none?

Reply:
Thank you for the suggestions! I created a new smaller function to get the first image HDU.